_overviews/scala3-contribution/procedures-checklist.md

```diff
@@ -7,35 +7,134 @@ previous-page: procedures-testing
+prepare a [pull request][pull-request] with your changes. Assuming that the team
+is aware of what you've been working, here are some final steps that you'll want
+to keep in mind as you create your PR.
 
 ### 1. Sign the CLA
 
-Make sure you have signed the [Scala CLA][cla], if not, sign it.
+Make sure you have signed the [Scala CLA][cla]. If you have any questions about
+what this is and why it's required you can read further about it [here][cla].
 
-### 2: Is It Relevant?
+### 2. Make sure your work is on its own branch
 
-Before starting to work on a feature or a fix, it's good practice to ensure that:
-1. There is a ticket for your work in the project's [issue tracker][issues];
-2. The ticket has been discussed and there is desire for it to be implemented by the
-Scala 3 core maintainers.
+When submitting your pull request it's always best to ensure the branch name is
+unique to the changes you're working on. It's important not to submit your PR on
+your `main` branch as this blocks maintainers from making any changes to your PR
+if necessary.
 
 ### 3: Add Tests
+
 Add at least one test that replicates the problem in the issue, and that shows it is now resolved.
 
 You may of course add variations of the test code to try and eliminate edge cases.
 [Become familiar with testing in Scala 3][testing].
 
 ### 4: Add Documentation
+
 Please ensure that all code is documented to explain its use, even if only internal
-changes are made.
+changes are made. This refers to scaladocs and also any changes that might be
+necessary in the reference docs.
+
+### 5: Double check everything
+
+Here are a couple tips to keep in mind.
+
+- [DRY (Don't Repeat Yourself)][dry]
+- [Scouts Rule][scouts]
+- When adding new code try use [optional braces]. If you're rewriting old code,
+  you should also use optional braces unless it introduces more code changes
+  that necessary.
+
+### 6: Commit Messages
+
+Here are some guidelines when writing commits for Dotty.
+
+1. If your work spans multiple local commits (for example; if you do safe point
+   commits while working in a feature branch or work in a branch for long time
+   doing merges/rebases etc.) then please do not commit it all but rewrite the
+   history by squashing the commits into one large commit which is accompanied
+   by a detailed commit message for (as discussed in the following sections).
+   For more info, see the article: [Git Workflow][git-workflow]. Additionally,
+   every commit should be able to be used in isolation—that is, each commit must
+   build and pass all tests.
+
+2. The first line should be a descriptive sentence about what the commit is
+   doing. It should be possible to fully understand what the commit does by just
+   reading this single line. It is **not ok** to only list the ticket number,
+   type "minor fix" or similar. If the commit has a corresponding ticket,
+   include a reference to the ticket number, prefixed with "Closes #", at the
```

Comment:
So maybe more of a process question. Is this done for specific reason? Like is there any sort of automation set up that needs this prefixed. The only reason I ask is that if you follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/), which many projects do, this is actually supposed to be in the footer. I realize I used the footer method in my example so I either need to change my example or change this. Maybe @anatoliykmetyuk you know more about the Dotty process?

Reply:
The reason we put the reference to the issue being fixed as the first line is purely historical. To my awareness, we do not follow the conventional commits format you linked.

Reply:
Alright, then I'll just leave this as is. If there is no process, then slightly pushing people to a format this is followed elsewhere is probably a good idea.

Thanks!